torch/distributed/_sharded_tensor/api.py

```diff
@@ -170,25 +163,21 @@ def __init__(
         self,
         sharding_spec: ShardingSpec,
         *size,
-        dtype=None,
-        layout=torch.strided,
-        requires_grad=False,
-        pin_memory=False,
-        memory_format=torch.contiguous_format,
+        common_params: InitCommonParams,
```

Comment:
is `_internals` namespace the right place for `InitCommonParams` given that its actually a public class to the user?

Reply:
Good point, this seems an interesting discussion. From existing code, I found: 
1. there was _internals.py under _sharding_spec 
2.the dir name _sharded_tensor and _sharding_spec suggested the entire folders are still internal 
3. the actual API interfaces seems living inside _sharded_tensor.__init__ (since the main APIs we wanted to expose is _sharded_tensor.empty, .ones, .init_from_local_sharded). the _sharded_tensor/api.py seems impl details.

How about we 
a) move _sharded_tensor/internals.py to _sharded_tensor/api.py 
and b) rename _sharded_tensor/api.py to sth like _sharded_tensor/_api.py or api_impl.py or sth else?

@rohan-varma @pritamdamania

Reply:
I kept the entire directory to be `_sharded_tensor` and `_sharding_spec` just as an indication that this is still initial prototype/beta work. Eventually we will rename these folders to be `sharding_spec` and `sharded_tensor` for an official release.

I think we can keep `InitCommonParams` in `api.py` itself but keep methods like `create_tensor_from_params` in internals.py

Reply:
After a 2nd thought, seems I need to remove _internals.py if we want to keep InitCommonParams (TensorInitParams) in api.py and create_tensor_from_params since that introduced circular dependency.